scipy/stats/tests/test_morestats.py

```diff
@@ -2873,105 +2873,107 @@ def test_result(self, correction):
         res = stats.median_test(x, y, correction=correction)
         assert_equal((res.statistic, res.pvalue, res.median, res.table), res)
 
-
+@array_api_compatible
 class TestDirectionalStats:
     # Reference implementations are not available
-    def test_directional_stats_correctness(self):
+    def test_directional_stats_correctness(self, xp):
         # Data from Fisher: Dispersion on a sphere, 1953 and
         # Mardia and Jupp, Directional Statistics.
-
-        decl = -np.deg2rad(np.array([343.2, 62., 36.9, 27., 359.,
-                                     5.7, 50.4, 357.6, 44.]))
-        incl = -np.deg2rad(np.array([66.1, 68.7, 70.1, 82.1, 79.5,
-                                     73., 69.3, 58.8, 51.4]))
-        data = np.stack((np.cos(incl) * np.cos(decl),
-                         np.cos(incl) * np.sin(decl),
-                         np.sin(incl)),
+        # angles were converted to radians
+        decl = xp.asarray([-5.98996999, -1.08210414, -0.64402649, -0.4712389,
+                           -6.26573201, -0.09948377, -0.87964594, -6.24129741,
+                           -0.76794487])
+        incl = xp.asarray([-1.15366264, -1.1990412 , -1.22347581, -1.43291532,
+                           -1.38753676, -1.27409035, -1.20951317, -1.0262536,
+                           -0.89709924])
```

Comment:
It is still allowed to use `np` within tests. It would reduce the diff here to leave the original definitions of `dec1` and `inc1` alone.

Do you have time to come back to this in July @dschmitz89?

Reply:
Thanks for the offer @ mdhaber. In the last four weeks I have not found time for any SciPy coding and the coming weeks are probably going to be similar. My SciPy contributions are at the moment rather limited to small comments and reviews. If anyone wants to take this one over, feel free to do so :).

Reply:
No problem @dschmitz89. 

@lucascolley do you have a moment to finish this up?